### PR TITLE
Add transmission class and examples

### PR DESCRIPTION
**Pull request by avrahamgoldman**

### CODE REVIEW

File: examples/bootstrap.php [hunk 1]
```diff
@@ -1,3 +1,7 @@
 <?php
 
 require_once dirname(__FILE__).'/../vendor/autoload.php';
+
+//pull in API key config
+$optionsFile = file_get_contents(dirname(__FILE__).'/example-options.json');
```
Comment (richleland) [inline examples/bootstrap.php]: We'll want to update the script in `composer.json` that creates this file as well.

File: examples/example-options.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+	"key": "YOUR_API_KEY"
```
Comment (richleland) [inline examples/example-options.json]: this file should be created by `composer.json`

File: examples/transmissions/create_transmission.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            'name' => 'SparkPost Team',
+            'email' => 'from@sparkpostbox.com'
+        ],
+        'subject'=>'First Mailing From PHP',
+        'html'=>'<html><body><h1>Congratulations, {{name}}!</h1><p>You just sent your very first mailing!</p></body></html>',
+        'text'=>'Congratulations, {{name}}!! You just sent your very first mailing!'
+    ],
+    'substitution_data'=> ['name'=>'YOUR_FIRST_NAME'],
+    'recipients'=> [
+        [ 'address' => 'YOUR_NAME <YOUR_EMAIL>' ]
```
Comment (richleland) [inline examples/transmissions/create_transmission.php]: Let's use the expanded name + email form here - I know we have parsing around this but this just pushes them towards using the better format.

File: examples/transmissions/create_transmission_with_cc_and_bcc.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            'name' => 'SparkPost Team',
+            'email' => 'from@sparkpostbox.com'
+        ],
+        'subject'=>'First Mailing From PHP',
+        'html'=>'<html><body><h1>Congratulations, {{name}}!</h1><p>You just sent your very first mailing!</p></body></html>',
+        'text'=>'Congratulations, {{name}}!! You just sent your very first mailing!'
+    ],
+    'substitution_data'=> ['name'=>'YOUR_FIRST_NAME'],
+    'recipients'=> [
+        [ 'address' => 'YOUR_NAME <YOUR_EMAIL>' ]
```
Comment (richleland) [inline examples/transmissions/create_transmission_with_cc_and_bcc.php]: Use expanded format for recipients, cc, and bcc.

File: lib/SparkPost/Resource.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * The api endpoint that gets prepended to all requests send through this resource.
+     */
+    protected $endpoint;
+
+    /**
+     * Sets up the Resource.
+     *
+     * @param SparKPost $sparkpost - the sparkpost instance that this resource is attached to
```
Comment (richleland) [inline lib/SparkPost/Resource.php]: lower case `K`

File: lib/SparkPost/Resource.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    public function delete($uri = '', $payload = [], $headers = [])
+    {
+        return $this->request('DELETE', $uri, $payload, $headers);
+    }
+
+    /**
+     * Sends requests to SparkPost object to the resource endpoint.
+     *
+     * @see SparkPost->request()
```
Comment (richleland) [inline lib/SparkPost/Resource.php]: do we need this here?

File: lib/SparkPost/Transmission.php [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+            } elseif (preg_match('/"?(.[^"]*)?"?\s*<(.+)>/', $address, $matches)) {
+                $name = trim($matches[1]);
+                $return['name'] = $matches[1];
+                $return['email'] = $matches[2];
+            } else {
+                throw new \Exception('Invalid address format: '.$address);
+            }
+        }
+
+        return $return;
```
Comment (richleland) [inline lib/SparkPost/Transmission.php]: the name of this var is a little weird - `$return` - maybe use `$formatted`?

**Comment by richleland** [comment]
Need to run `composer run-script fix-style`, which runs php-cs-fixer on the code for style.

**Comment by richleland** [comment]
Looks great thus far. Single recip, multiple recip, and cc/bcc all work.

File: examples/transmissions/create_transmission.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+try {
+    $response = $promise->wait();
+    echo $response->getStatusCode() . "\n";
+    print_r($response->getBody());
+} catch (Exception $e) {
+    echo $e->getCode() . "\n";
+    echo $e->getMessage() . "\n";
+}
+?>
```
Comment (zaporylie) [inline examples/transmissions/create_transmission.php]: Missing new line character at end of file

**Comment by richleland** [comment]
👌 - once the build completes, merge away!

**Pull request merged**